pkg/apis/core/validation/validation.go

```diff
@@ -3075,6 +3075,39 @@ func validateContainerCommon(ctr *core.Container, volumes map[string]core.Volume
 	allErrs = append(allErrs, validatePullPolicy(ctr.ImagePullPolicy, path.Child("imagePullPolicy"))...)
 	allErrs = append(allErrs, ValidateResourceRequirements(&ctr.Resources, path.Child("resources"), opts)...)
 	allErrs = append(allErrs, ValidateSecurityContext(ctr.SecurityContext, path.Child("securityContext"))...)
+	return allErrs
+}
+
+func validateHostUsers(spec *core.PodSpec, volumes map[string]core.VolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) {
+		if spec.SecurityContext != nil && spec.SecurityContext.HostUsers != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("hostUsers"), "field not supported. Enable the feature gate UserNamespacesSupport"))
+		}
+		return allErrs
+	}
+
+	// Only make the following checks if hostUsers is false (otherwise, the container uses the
+	// same userns as the host, and so there isn't anything to check).
+	if spec.SecurityContext == nil || spec.SecurityContext.HostUsers == nil || *spec.SecurityContext.HostUsers == true {
+		return allErrs
+	}
+
+	// For now only these volumes are supported:
+	// - configmap
+	// - secret
+	// - downwardAPI
+	// - emptyDir
+	// - projected
+	// So reject anything else.
+	for name, vol := range volumes {
+		if vol.EmptyDir != nil || vol.Secret != nil || vol.DownwardAPI != nil || vol.ConfigMap != nil ||
+			vol.Projected != nil {
+			continue
+		}
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child(name), "volume type not yet supported in combination with pod.Spec.HostUsers=false"))
```

Comment:
this doesn't tell the user what volume index wasn't supported... to point to the right field, I think you need `fldPath.Child("volumes").Index(i)`

Reply:
Changed to use the index. It was showing the volume name before, which is also nice